src/IPC/PipeTx.cc

```diff
@@ -1,21 +1,75 @@
 #include "IPC.h"
 
-void PipeTx::send(void) {
-    this->setupPipeTx();
+#include <sys/stat.h>   /* mkfifo() */
+#include <fcntl.h>      /* O_CREAT, O_WRONLY */
+#include <iostream>     /* std::cout std::ios seekg() */
+#include <unistd.h>     /* write() */
 
-    this->fileSizeTx();
+#define FIFO "./fifoChannel"
 
+void PipeTx::send(void) {
+    this->setupPipeTx();
+    
     this->pipeTx();
 }
 
 void PipeTx::setupPipeTx(void) {
-    ;
-}
-
-void PipeTx::fileSizeTx(void) {
-    ;
+    mkfifo(FIFO, 0666);                         /* read/write for user/group/others */
+    this->fd = open(FIFO, O_CREAT | O_WRONLY);  
+    if (fd < 0) {
+        throw std::runtime_error("Tx Pipe couldn't be opened");
+    }
+    std::cout << "Tx opened" << std::endl;     
 }
 
 void PipeTx::pipeTx(void) {
-    ;
+    int n = 0;  /* keeping track of number of full buffer sent */
+
+    /* determine length of file */
+    this->readFile->seekg(0, std::ios::beg);    /* set position to begin of file */
```

Comment:
There is some controversy about this method of finding file size. But anyway, you don't need file size at all. You can replace all usages of 'remainingBytes' by just checking 'read' return values

Reply:
The read function I'm using returns `std::istream`, to change this to the read function you're referring to it would mean also changing a lot of the functions used by the class. Is there another way to circumvent this issue? There has to be a 'good' way to find the file size, right? :)

![Screenshot from 2022-04-12 08-10-48](https://user-images.githubusercontent.com/97879338/162886157-930020a0-e070-4176-b83d-b2daa97feafd.png)

Reply:
right, but this returned ifstream has function 'gcount()', which will give just read number of bytes.
So it can be:
`readBytes = this->readFile->read(...).gcount()`
If you still prefer to find file size, you can use 'stat' function and take st_size attribute

Reply:
Ok, interesting, didn't know about this one. :+1: